test/test_reaction.py

```diff
@@ -227,19 +266,50 @@ def test_reactant1_setter_raises_error_with_wrong_type():
         )
 
 
-def test_reactant2_setter_raises_error_with_wrong_type():
-    """Test a type error is raised when the reactant2 is given a wrong type."""
+def test_reactant_setter_raises_error_with_wrong_type():
+    """Test a type error is raised when the first reactant is given a wrong type."""
     with pytest.raises(
         TypeError,
-        match="reactant2 must be an F.Species or F.ImplicitSpecies, not <class 'str'>",
+        match="reactant must be an F.Species or F.ImplicitSpecies, not <class 'str'>",
     ):
         F.Reaction(
-            reactant1=F.Species("A"),
-            reactant2="B",
+            reactant=["A", F.Species("B")],
             product=F.Species("C"),
             k_0=1,
             E_k=0.1,
             p_0=2,
             E_p=0.2,
             volume=my_vol,
         )
+
+
+def test_product_setter_raise_error_p_0_no_product():
+    with pytest.raises(
+        ValueError,
+        match="p_0 must be 0, not 2 when no products are present.",
+    ):
+        F.Reaction(
+            reactant=[F.Species("A")],
+            product=None,
+            k_0=1,
+            E_k=0.1,
+            p_0=2,
+            E_p=0,
+            volume=my_vol,
+        )
+
+
+def test_product_setter_raise_error_E_p_no_product():
+    with pytest.raises(
+        ValueError,
+        match="E_p must be 0, not 2 when no products are present.",
+    ):
+        F.Reaction(
+            reactant=[F.Species("A")],
+            product=None,
+            k_0=1,
+            E_k=0.1,
+            p_0=0,
+            E_p=2,
+            volume=my_vol,
+        )
```

Comment:
So here is a good example of why this could be more straightforward.

Here, if p_0 = 0, then E_p could be equal to any value and it wouldn't matter.

Therefore, I think it's more straightforward to have p_0 and E_p default to `None` in `Reaction` instead of making sure they are equal to zero.